jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/QoSHandler.java

```diff
@@ -353,15 +355,31 @@ private boolean resumeSuspended()
                 if (LOG.isDebugEnabled())
                     LOG.debug("{} resuming {}", this, entry.request);
                 // Always dispatch to avoid StackOverflowError.
-                getServer().getThreadPool().execute(entry);
+                execute(entry, entry.useVirtualThreads);
                 return true;
             }
         }
         return false;
     }
 
+    private void execute(Runnable task, boolean useVirtualThreads)
+    {
+        ThreadPool executor = getServer().getThreadPool();
+        if (useVirtualThreads)
+        {
+            Executor virtualExecutor = VirtualThreads.getVirtualThreadsExecutor(executor);
+            if (virtualExecutor != null)
+            {
+                virtualExecutor.execute(task);
+                return;
+            }
+        }
+        executor.execute(task);
+    }
```

Comment:
I don't think we should do specific QoSHandler resolution for this.  See my comments on #12171.